app/services/import_comment_service.rb

```diff
@@ -0,0 +1,24 @@
+class ImportCommentService < ApplicationService
+  def initialize(post, comments)
+   @post = post
+   @comments = comments
+  end
+
+  def call
+    comments.each do |comment|
+      fid = comment['id']
+      Comment.find_or_create_by(fid: fid) do |c|
+        c.fid = fid
+        c.message = comment['message']
+        c.created_at = comment['created_time']
+        c.user = ImportUserService.call(comment['from'])
+        c.post = post
+      end
```

Comment:
so instead of find_or_create_by we can use find_or_initialize, map all comments to objects (not saved yet), then assign them to `post.comments` probably that would be faster? anyway it's worth a try.

also wrapping this call in a transaction may help make it faster?

Reply:
yeah . it's very slow, took me a lot more than one hour

i think inserting the post comments all at once will indeed enhance the performance, let me try it,
also i think we can skip uniqueness validation on Rails level and depend only on the db unique index ?, this will prevent the query rails do before every creation to check if it's already there.
i'm not sure about the transaction point though